xmr-btc/src/bob.rs

```diff
@@ -660,51 +660,34 @@ impl State4 {
         self.b.encsign(self.S_a_bitcoin, tx_redeem.digest())
     }
 
-    pub async fn check_for_tx_cancel<W>(&self, bitcoin_wallet: &W) -> Result<Transaction>
-    where
-        W: GetRawTransaction,
-    {
-        let tx_cancel =
-            bitcoin::TxCancel::new(&self.tx_lock, self.refund_timelock, self.A, self.b.public());
-
-        let sig_a = self.tx_cancel_sig_a.clone();
-        let sig_b = self.b.sign(tx_cancel.digest());
-
-        let tx_cancel = tx_cancel
-            .clone()
-            .add_signatures(&self.tx_lock, (self.A, sig_a), (self.b.public(), sig_b))
-            .expect(
-                "sig_{a,b} to be valid signatures for
-                tx_cancel",
-            );
-
-        let tx = bitcoin_wallet.get_raw_transaction(tx_cancel.txid()).await?;
-
-        Ok(tx)
+    pub fn tx_cancel(&self) -> TxCancel {
+        crate::bitcoin::TxCancel::new(&self.tx_lock, self.refund_timelock, self.A, self.b.public())
     }
 
     pub async fn submit_tx_cancel<W>(&self, bitcoin_wallet: &W) -> Result<Txid>
     where
         W: BroadcastSignedTransaction,
     {
-        let tx_cancel =
-            bitcoin::TxCancel::new(&self.tx_lock, self.refund_timelock, self.A, self.b.public());
+        let tx_cancel = self.tx_cancel();
 
-        let sig_a = self.tx_cancel_sig_a.clone();
-        let sig_b = self.b.sign(tx_cancel.digest());
+        let sig_theirs = self.tx_cancel_sig_a.clone();
```

Comment:
```suggestion
        let sig_theirs = self.tx_cancel_sig_alice.clone();
```

❗️ This code is becoming more and more confusing because different terminologies are mixed together.
Above you refer to `tx_punish_sig_bob` at this point, should this be `tx_cancel_sig_alice`?
Please use the same terminology throughout to avoid confusion.

Reply:
Removed theirs and hours

I can change to `tx_cancel_sig_alice`. I didn't make it `tx_cancel_sig_a`, it was already like that.